core/src/main/java/com/datastrato/gravitino/catalog/TableOperationDispatcher.java

```diff
@@ -79,37 +82,25 @@ public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaExcep
    */
   @Override
   public Table loadTable(NameIdentifier ident) throws NoSuchTableException {
-    NameIdentifier catalogIdentifier = getCatalogIdentifier(ident);
-    Table table =
-        doWithCatalog(
-            catalogIdentifier,
-            c -> c.doWithTableOps(t -> t.loadTable(ident)),
-            NoSuchTableException.class);
+    EntityCombinedTable table =
+        TreeLockUtils.doWithTreeLock(ident, LockType.READ, () -> loadCombinedTable(ident));
```

Comment:
I noticed that you scattered the logic for determining whether it is `imported` in and after the `loadCombinedTable` method. 

Can you consolidate the code for judging `imported` after `loadCombinedTable`? This can make the responsibilities of the method clearer. 

Or maybe we can put the logic related to `imported` into `XxxNormalizeDispatcher` class and handle it as part of the normalization operation?

Reply:
1.No, we can't. It will use a bigger lock.
2. I prefer not. It's unrelated to NormalizeDispatcher.